src/Angor/Client/Pages/Spend.razor

```diff
@@ -781,12 +801,49 @@
                 {
                     item.IsSpent = true;
 
-                    updateUnconfirmedOutbound |= LinqExtension.Remove(unconfirmedOutbound, new Outpoint(item.Trxid, item.Outputindex));
+                    updateUnconfirmedOutbound |= unconfirmedOutbound.TryRemoveOutpoint(new Outpoint(item.Trxid, item.Outputindex));
+
+                    // try to resolve the destination
+                    var spentInTransaction = await _IndexerService.GetTransactionInfoByIdAsync(output.SpentInTransaction);
+
+                    var input = spentInTransaction?.Inputs.FirstOrDefault(input => input.InputTransactionId == item.Trxid && input.InputIndex == item.Outputindex);
+
+                    var investmentTransaction = trxsHexCache.TryGet(item.Trxid);
+                    if (investmentTransaction == null)
+                    {
+                        investmentTransaction = network.CreateTransaction(await _IndexerService.GetTransactionHexByIdAsync(item.Trxid));
+                        trxsHexCache.Add(item.Trxid, investmentTransaction);
+                    }
+
+                    if (input != null && investmentTransaction != null)
+                    {
+                        item.ProjectScriptType = _InvestorTransactionActions.DiscoverUsedScript(project, investmentTransaction, stageData.StageIndex - 1, input.WitScript);
+
+                        switch (item.ProjectScriptType.ScriptType)
+                        {
+                            case ProjectScriptTypeEnum.Founder:
+                            {
+                                item.SpentType = "founder";
+                                break;
+                            }
+                            case ProjectScriptTypeEnum.InvestorWithPenalty:
+                            case ProjectScriptTypeEnum.EndOfProject:
+                            case ProjectScriptTypeEnum.InvestorNoPenalty:
+                            {
+                                item.SpentType = "investor";
+                                break;
+                            }
+                        }
+                    }
 
                     continue;
                 }
 
-                item.IsSpent = LinqExtension.Contains(unconfirmedOutbound, new Outpoint(item.Trxid, item.Outputindex));
+                item.IsSpent = unconfirmedOutbound.ContainsOutpoint(new Outpoint(item.Trxid, item.Outputindex));
+                if (item.IsSpent)
+                {
+                    item.SpentType = "pending";
```

Comment:
Can you add a comment so it's clear what pending means? is it not related to who spent it?

Reply:
To solve who spent pending we would need to fetch data of unconfirmed trx, it may be possible but I think it is best to do it in a separate pr because we dont really try to resolve pending trx anywhere